### PR TITLE
[docs] Code inject

### PR DESCRIPTION
**Pull request by ronny-mysten**
## Description 

Updates the code inject plugin to capture move functions, structs, and modules by name. Moves coin and nft content up a level to improve discoverability. 

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: examples/move/first_package/Move.toml [hunk 1]
```diff
@@ -1,10 +1,36 @@
...
-version = "0.0.1"
-edition = "2024.beta"
+name = "my_first_package"
+edition = "2024.beta" # edition = "legacy" to use legacy (pre-2024) Move
+# license = ""           # e.g., "MIT", "GPL", "Apache 2.0"
+# authors = ["..."]      # e.g., ["Joe Smith (joesmith@noemail.com)", "John Snow (johnsnow@noemail.com)"]
 
 [dependencies]
-Sui = { local = "../../../crates/sui-framework/packages/sui-framework" }
+Sui = { git = "https://github.com/MystenLabs/sui.git", subdir = "crates/sui-framework/packages/sui-framework", rev = "framework/testnet" }
```
[the remark this answers is not in the document]
Comment (ronny-mysten) [reply examples/move/first_package/Move.toml:8]: I didn't do this intentionally. I think I was copy/pasting from (what I thought was) an updated source. Will look into it.

File: docs/site/src/plugins/inject-code/utils.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    } else {
+      pre.push(lines[x].substring(numSpaces));
+    }
+  }
+  return pre.reverse().join("\n");
+};
+
+// If opening and closing braces don't match
+// there's a problem
+exports.checkBracesBalance = (str) => {
```
[the remark this answers is not in the document]
Comment (ronny-mysten) [reply docs/site/src/plugins/inject-code/utils.js:104]: Yeah, it's already proved to be a pita. But the idea is to act as kind of a safety net. In the end, if it fails to return 'true', the author will just have to use the doc comment approach instead. I'll add that note to the documentation.

[the remark this answers is not in the document]
Comment (amnn) [reply docs/site/src/plugins/inject-code/utils.js:104]: I'm curious -- how often has it helped vs. hurt? (In how many cases has it legitimately prevented an issue?) If it's not many, then it may also be worth dropping the check as well...

File: docs/content/concepts/dynamic-fields.mdx [hunk 3]
```diff
@@ -114,24 +65,11 @@ Where `object` is the UID of the object the field is defined on and `name` is th
...
-}
-
-public fun mutate_child_via_parent(parent: &mut Parent) {
-    mutate_child(ofield::borrow_mut(
-        &mut parent.id,
-        b"child",
-    ));
-}
-```
+{@inject: examples/move/dynamic_fields/sources/example.move#fun=mutate_child,mutate_child_via_parent noComments}
```
[the remark this answers is not in the document]
Comment (ronny-mysten) [reply docs/content/concepts/dynamic-fields.mdx:68]: Grr. I fix one thing just to break another...

File: docs/site/src/plugins/inject-code/utils.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ronny-mysten) [reply docs/site/src/plugins/inject-code/utils.js:104]: Good question. I think the original reason was because it would grab the module closing brace if the reference was for the last item and I couldn't get a proper regex to filter it out in those cases. Then I thought it might be a good, basic check. But I think there's probably a better approach for that one case and just use the fallback for any corners that might come up.

[the remark this answers is not in the document]
Comment (ronny-mysten) [reply docs/site/src/plugins/inject-code/utils.js:128]: This remains but only as a function of processing the last thing in a block/module. Not used as a check.

File: docs/content/guides/developer/nft.mdx [hunk 4]
```diff
@@ -0,0 +1,41 @@
+---
+title: Create a Non-Fungible Token
+keywords: [ERC-721, NFT]
+---
+
+On Sui, everything is an object. Moreover, everything is a non-fungible token (NFT) as its objects are unique, non-fungible, and owned. So technically, a basic type publishing is enough to create a specific NFT.
```
Comment (0xaslan) [inline docs/content/guides/developer/nft.mdx:6]: `So technically, a basic type publishing is enough to create a specific NFT.` Had to reread this a few times. Can it be reworded or omitted?

Comment (ronny-mysten) [reply docs/content/guides/developer/nft.mdx:6]: Good call

File: examples/move/coin/sources/regcoin.move [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+//docs::#regulate
+module examples::regcoin {
+    use sui::coin::{Self, DenyCap};
+    use sui::deny_list::{DenyList};
+
+    public struct REGCOIN has drop {}
+
+    fun init(witness: REGCOIN, ctx: &mut TxContext) {
+        let (treasury, deny_cap, metadata) = coin::create_regulated_currency(witness, 6, b"REGCOIN", b"", b"", option::none(), ctx);
+        transfer::public_freeze_object(metadata);
```
Comment (manolisliolios) [inline examples/move/coin/sources/regcoin.move:13]: I am not convinced we should recommend freezing metadata.
Applies to `my_coin.move` too.

**Review by manolisliolios — APPROVED** [review]
Approved to unblock!

File: Cargo.lock [hunk 6]
```diff
@@ -6279,6 +6279,15 @@ dependencies = [
...
 
+[[package]]
+name = "lru"
+version = "0.12.3"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "d3262e75e648fce39813cb56ac41f3c3e3f65217ebf3844d818d1f9398cfb0dc"
+dependencies = [
+ "hashbrown 0.14.1",
+]
+
```
Comment (manolisliolios) [inline Cargo.lock:6290]: Why is this needed?

File: examples/move/coin/sources/regcoin.move (hunk 5 above)
Comment (ronny-mysten) [reply examples/move/coin/sources/regcoin.move:13]: If you're not, I'm definitely not.

File: Cargo.lock (hunk 6 above)
Comment (ronny-mysten) [reply Cargo.lock:6290]: hashbrowns? Needed because they're delicious. Not sure how this got in here, though. Removing...
